adm/unix/compileunix.py

```diff
@@ -31,28 +26,29 @@ def pathExists(path):
 def gcc(exeFile, srcFiles, includes, libs, defs):
   # standard includes
   cmd = "gcc"
-  if is64bit():
-      cmd += " -m32" # always compile in 32bit mode
+  if (platform.machine() == 'x86_64'):
+    cmd += " -m32" # always compile in 32bit mode
 
   for include in includes:
-    cmd += " -I\"" + include + "\""
+    cmd += " -I" + include
 
   # defines (tuples)
   for d in defs:
-    cmd += " -D" + d[0] + "=" + d[1]
+    if not isinstance(d, tuple):
+      d = (d, )
+    cmd += " -D" + d[0]
+    if len(d)>1 and d[1] is not None:
+      cmd += "=" + d[1]
 
   cmd += " -DPLAT_BUILD_VERSION=" + '\\"' + env.buildVersion() + '\\"'
 
   # src
   for src in srcFiles:
     cmd += " " + src
 
-  # lib paths
+  # libs
   for lib in libs:
-    if pathExists(lib):
```

Comment:
with path existence check, we can support both '-l' and '-L', what do you think ?

Reply:
I think it's not a flexibility, but another hardcoded option. I'd place -L<path> in the same string with a library.  And after library name any compile/link option needed for this library to work can be placed, including library path.
Honestly I don't like these hardcoded -D and -l. What about other needed CFLAGS, LDFLAGS, like a '-pthread'? It's not a define and not a library name.
I'm going to introduce submitting extra CFLAGS and LDFLAGS using environment variables in the next pull request. So it's not necessary to patch makeunixvm.py every time some external native kit. What do you think?

Reply:
yeah, agreed. I think a more suitable method to handle compilation is makefile, not python script. but since the svm compilation normally should be simple, just live with this python build scripts now, and put efforts on the more interesting parts first.